src/msg/async/net_handler.cc

```diff
@@ -126,8 +129,6 @@ int NetHandler::generic_connect(const entity_addr_t& addr, bool nonblock)
     return -errno;
   }
 
-  set_socket_options(s);
-
```

Comment:
@yuyuyu101 @tchaikov 
after testing(ss command line), I found that SO_RCVBUF has no effect, and we should set it before connect or listen.

man 7 tcp, get the information:
The maximum sizes for socket buffers declared via the SO_SNDBUF and SO_RCVBUF mechanisms are limited by the values in the /proc/sys/net/core/rmem_max and /proc/sys/net/core/wmem_max  files.
Note  that  TCP  actually  allocates  twice  the  size  of  the buffer requested in the setsockopt(2) call, and so a succeeding getsockopt(2) call will not return the same size of buffer as requested in the setsockopt(2) call.  TCP uses the extra space for administrative purposes and internal kernel structures, and the /proc file values reflect the larger sizes compared to the actual TCP windows.  On individual connections, **the socket buffer size must be set prior to the listen(2) or connect(2) calls in order to have it take effect.**  See socket(7) for more information.